web/views.py

```
@@ -7018,3 +7024,577 @@ def users_list(request: HttpRequest) -> HttpResponse:
+def chapter_detail(request, slug):
+    """Show details of a specific chapter."""
+    chapter = get_object_or_404(Chapter, slug=slug, is_active=True)
+    
+    # Check if user is a member and get their role
+    user_membership = None
+    if request.user.is_authenticated:
+        user_membership = chapter.members.filter(user=request.user).first()
+    
+    # Get upcoming events
+    upcoming_events = chapter.events.filter(
+        start_time__gte=timezone.now()
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)